src/cargo/core/compiler/mod.rs

```diff
@@ -1028,6 +1030,10 @@ fn build_base_args(cx: &Context<'_, '_>, cmd: &mut ProcessBuilder, unit: &Unit)
         }
     }
 
+    if let Some(trim_paths) = trim_paths {
+        trim_paths_args(cmd, cx, unit, &trim_paths)?;
+    }
```

Comment:
Should this also be included in rustdoc to handle the diagnostics stripping (and same with doctests)?

(Though I still don't 100% understand the purpose of diagnostic remapping.)

Reply:
I don't think there is such an option for `rustdoc`. Did I miss something? 
Maybe this could be put in "unresolved questions" in RFC 3127 tracking issue.

```console
$ rustdoc +nightly -vV
rustdoc 1.75.0-nightly (608e9682f 2023-10-29)
binary: rustdoc
commit-hash: 608e9682f0a6482903de8d3332770104a0ad943c
commit-date: 2023-10-29
host: aarch64-apple-darwin
release: 1.75.0-nightly
LLVM version: 17.0.3

$ rustc --remap-path-prefix
error: Argument to option 'remap-path-prefix' missing

$ rustdoc --remap-path-prefix
error: Unrecognized option: 'remap-path-prefix'
```

Reply:
Yea, leaving that as an open issue sounds good. The rustdoc argument parser would just need to be updated.

Though, before that I would like to better understand why the diagnostic option exists and when someone would want to turn it on. I can't really follow the discussion from https://github.com/rust-lang/rust/issues/87745 and how that relates to cargo. If we end up not stabilizing "diagnostic" as an option, then I don't have any motivation to do that.

Reply:
Regrading the `diagnostics` option I think it's related to build-systems that create VFS (Virtual File System) or temporary location when building, throwing the all diagnostics path to the wrong location.